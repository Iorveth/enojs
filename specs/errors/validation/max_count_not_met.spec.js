const eno = require('../../../eno.js');

const input = `
languages:
- cson
- eno
- json
- yaml
`.trim();

describe('validation.maxCountNotMet', () => {
  const document = eno.parse(input);

  let error;
  try {
    document.list('languages', { maxCount: 3 });
  } catch(err) {
    error = err;
  }

  it(`provides a correct message`, () => {
    expect(error.message).toMatchSnapshot();
  });

  it(`provides correct selection metadata`, () => {
    expect(error.selection).toMatchSnapshot();
  });
});
