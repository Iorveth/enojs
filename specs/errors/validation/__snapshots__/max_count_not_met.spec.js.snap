// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation.maxCountNotMet provides a correct message 1`] = `
"The list 'languages' contains 4 items, but may only contain a maximum of 3 items.

   Line | Content
 *    1 | languages:
 >    2 | - cson
 >    3 | - eno
 >    4 | - json
 >    5 | - yaml
"
`;

exports[`validation.maxCountNotMet provides correct selection metadata 1`] = `
Array [
  Array [
    1,
    0,
  ],
  Array [
    4,
    6,
  ],
]
`;
